R/wlr.R

```diff
@@ -75,44 +86,86 @@
 #'   \deqn{z = \sum_i X_i/\sqrt{\sum_i V_i}.}
 #'
 #' @examples
+#' # ---------------------- #
+#' #      Example 1         #
+#' #  Use dataset generated #
+#' #     by simtrial        #
+#' # ---------------------- #
 #' x <- sim_pw_surv(n = 200) |> cut_data_by_event(100)
 #'
-#' # Example 1: WLR test with FH wights
+#' # Example 1A: WLR test with FH wights
 #' x |> wlr(weight = fh(rho = 0, gamma = 0.5))
 #' x |> wlr(weight = fh(rho = 0, gamma = 0.5), return_variance = TRUE)
 #'
-#' # Example 2: WLR test with MB wights
+#' # Example 1B: WLR test with MB wights
 #' x |> wlr(weight = mb(delay = 4, w_max = 2))
 #'
-#' # Example 3: WLR test with early zero wights
+#' # Example 1C: WLR test with early zero wights
 #' x |> wlr(weight = early_zero(early_period = 4))
 #'
+#' # Example 1D
 #' # For increased computational speed when running many WLR tests, you can
 #' # pre-compute the counting_process() step first, and then pass the result of
 #' # counting_process() directly to wlr()
 #' x <- x |> counting_process(arm = "experimental")
 #' x |> wlr(weight = fh(rho = 0, gamma = 1))
 #' x |> wlr(weight = mb(delay = 4, w_max = 2))
 #' x |> wlr(weight = early_zero(early_period = 4))
-wlr <- function(data, weight, return_variance = FALSE) {
+#'
+#' # ---------------------- #
+#' #      Example 2         #
+#' #  Use cumsum dataset    #
+#' # ---------------------- #
+#' x <- data.frame(treatment = ifelse(ex1_delayed_effect$trt == 1, "experimental", "control"),
+#'                 stratum = rep("All", nrow(ex1_delayed_effect)),
+#'                 tte = ex1_delayed_effect$month,
+#'                 event = ex1_delayed_effect$evntd)
+#' class(x) <- c("tte_data", class(x))
+#'
+#' # Users can specify the randomization ratio to calculate the statistical information under H0
+#' x |> wlr(weight = fh(rho = 0, gamma = 0.5), ratio = 2)
+#'
+#' x |>
+#'   counting_process(arm = "experimental") |>
+#'   wlr(weight = fh(rho = 0, gamma = 0.5), ratio = 2)
+#'
+#' # If users don't provide the randomization ratio, we will calculate the emperical ratio
+#' x |> wlr(weight = fh(rho = 0, gamma = 0.5))
```

Comment:
Could we please convert this example to an automated test so that we can confirm that manually providing `ratio = 2` returns a different result compared to using the empirical ratio via `ratio = NULL`?

Reply:
Hello @jdblischak, I appreciate you bringing this to my attention. I completely agree with you that adding tests is valuable and advantageous. But I tentatively prefer to keep it as it is. My reasons are listed as follows.

First of all, since there is no equivalence between the one with `ratio=2` and `ratio=NULL`, it's challenging to compare these two on the same scale. 

Second, rather than composing my own developer tests, I should seek volunteers for independent or double programming tests, which hold more value and solidity compared to my basic developer tests.

Third, instead of including the above in the tests, I would rather keep them as examples in the help page, where users can readily distinguish the two ways to set `ratio`, even if they don't review the test files.

Reply:
> I completely agree with you that adding tests is valuable and advantageous.

I'll add the tests in a follow-up PR

> First of all, since there is no equivalence between the one with `ratio=2` and `ratio=NULL`, it's challenging to compare these two on the same scale. 

I don't understand the issue with the scale. I am requesting a test that confirms that the newly added argument `ratio` has an effect on the results. Thus something like the below would be sufficient:

```R
wlr_w_ratio <- x |> wlr(weight = fh(rho = 0, gamma = 0.5), ratio = 2)
wlr_wo_ratio <- x |> wlr(weight = fh(rho = 0, gamma = 0.5))
expect_false(isTRUE(all.equal(wlr_w_ratio, wlr_wo_ratio)))

```

> Second, rather than composing my own developer tests, I should seek volunteers for independent or double programming tests, which hold more value and solidity compared to my basic developer tests.

I do not understand this perspective. The potential for better tests in the future does not remove the requirement for basic developer tests. Tests are our best method to prevent the accumulation of [technical debt](https://en.wikipedia.org/wiki/Technical_debt). As a concrete, recent example, the argument `ratio` of `gsDesign2::fixed_design_mb()` had no effect because it was not passed to the downstream sub-functions (https://github.com/Merck/gsDesign2/pull/463). A basic developer test could have caught this.

> Third, instead of including the above in the tests, I would rather keep them as examples in the help page, where users can readily distinguish the two ways to set `ratio`, even if they don't review the test files.

I was not requesting that the examples be removed. I completely agree with you that the examples section is the correct place to document this for end users. My request was to use these examples as the basis for automated tests.